src/AWS.jl

```diff
@@ -33,19 +33,39 @@ aws_config = Ref{AWSConfig}()
     global_aws_config()
 
 Retrieve the global AWS configuration.
+If one is not set, create one with default configuration options.
+
+# Keywords
+- `kwargs...`: AWSConfig kwargs to be passed along if the global configuration is not already set
 
 # Returns
 - `AWSConfig`: The global AWS configuration
 """
-function global_aws_config()
+function global_aws_config(; kwargs...)
     if !isassigned(aws_config)
-        aws_config[] = AWSConfig()
+        aws_config[] = AWSConfig(; kwargs...)
     end
 
     return aws_config[]
 end
 
 
+"""
+    set_global_aws_config(config::AWSConfig)
+
+Set the global AWSConfig.
+
+# Arguments
+- `config::AWSConfig`: The AWSConfig to set in the global state
+
+# Returns
+- `AWSConfig`: Global AWSConfig
+"""
+function set_global_aws_config(config::AWSConfig)
```

Comment:
Doesn't need to be a different function name. I you really want a different function name I would suggest `global_aws_config!`:
```suggestion
function global_aws_config(config::AWSConfig)
```